repositories/convoy_repository.go

```diff
@@ -114,3 +114,93 @@ func (repo ConvoyRepository) RegisterWebhook(ctx context.Context, input models.W
 
 	return nil
 }
+
+func (repo ConvoyRepository) ListWebhooks(ctx context.Context, organizationId string, partnerId null.String) ([]models.Webhook, error) {
+	projectId := repo.convoyClientProvider.GetProjectID()
+	convoyClient, err := repo.convoyClientProvider.GetClient()
+	if err != nil {
+		return nil, err
+	}
+
+	ownerId := getOwnerId(organizationId, partnerId)
+
+	endpoints, err := convoyClient.GetEndpointsWithResponse(ctx, projectId, &convoy.GetEndpointsParams{
+		OwnerId: &ownerId,
+		PerPage: utils.Ptr(100),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "can't get convoy endpoints: request error")
+	}
+	if endpoints.JSON200 == nil {
+		return nil, errors.New("can't get convoy endpoints: response error")
+	}
+
+	endpointMap := make(map[string]convoy.ModelsEndpointResponse)
+	endpointIds := make([]string, 0, len(*endpoints.JSON200.Data.Content))
+	for _, convoyEndpoint := range *endpoints.JSON200.Data.Content {
+		endpointIds = append(endpointIds, *convoyEndpoint.Uid)
+		endpointMap[*convoyEndpoint.Uid] = convoyEndpoint
+	}
+
+	subscriptions, err := convoyClient.GetSubscriptionsWithResponse(ctx, projectId, &convoy.GetSubscriptionsParams{
+		EndpointId: &endpointIds,
+		PerPage:    utils.Ptr(100),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "can't get convoy subscriptions: request error")
+	}
+	if subscriptions.JSON200 == nil {
+		return nil, errors.New("can't get convoy subscriptions: response error")
+	}
+
+	webhooks := make([]models.Webhook, 0, len(*subscriptions.JSON200.Data.Content))
+	for _, convoySubscription := range *subscriptions.JSON200.Data.Content {
+		convoyEndpoint, ok := endpointMap[*convoySubscription.EndpointMetadata.Uid]
+		if !ok {
+			return nil, errors.New("can't find convoy endpoint")
+		}
+
+		webhooks = append(webhooks, adaptWebhook(convoyEndpoint, convoySubscription))
+	}
+
+	return webhooks, nil
+}
```

Comment:
Due to the opinionated way we handle webhook on Convoy (create imediately a pair endpoint/subscription) I need to fetch and reconcile both entities on Convoy (to return something close to what we expect for webhook register action).

An alternative would be :
- `LIST` : fetch only endpoints and display part of the data (= in a list view) 
- `GET /:id` : get full info per endpoint  (= with dedicated detail view)

Still, I think we can live with that, assuming we won't create too many webhooks in the near future (-> that's why I hardcoded a 100 page size and do not bother with pagination)

Reply:
I can live with the setup where we retrieve both pieces of information and merge them.
We'll see in due time if it ever becomes an issue.
For good measure, let's use a larger page size and/or log a warning if nb of items retrieved == page size ?